### PR TITLE
fix(tabs): adjust the watch execution order of tabs and activeIndex

### PR DESCRIPTION
**Pull request by Saber-Kurama**
## Types of changes

- [ ] New feature
- [x] Bug fix
- [ ] Enhancement
- [ ] Component style change
- [ ] Typescript definition change
- [ ] Documentation change
- [ ] Coding style change
- [ ] Refactoring
- [ ] Test cases
- [ ] Continuous integration
- [ ] Breaking change
- [ ] Others

## Changelog

| Component | Changelog(CN) | Changelog(EN) | Related issues |
| --------- | ------------- | ------------- | -------------- |
|     tabs      |        调整 tabs 和 activeIndex 的 watch 执行顺序    |      adjust the watch execution order of tabs and activeIndex         |          #1867      |

## Checklist:

- [ ] Test suite passes (`npm run test`)
- [ ] Provide changelog for relevant changes (e.g. bug fixes and new features) if applicable.
- [ ] Changes are submitted to the appropriate branch (e.g. features should be submitted to `feature` branch and others
  should be submitted to `main` branch)

### CODE REVIEW

File: packages/web-vue/components/tabs/tabs.tsx [hunk 1]
```diff
@@ -329,9 +329,9 @@ export default defineComponent({
           {mergedPosition.value === 'bottom' && renderContent()}
           <TabsNav
             v-slots={{ extra: slots.extra }}
-            tabs={sortedTabs.value}
             activeKey={computedActiveKey.value}
             activeIndex={activeIndex.value}
+            tabs={sortedTabs.value}
```
Comment (flsion) [inline packages/web-vue/components/tabs/tabs.tsx:334]: 仅修改 tabs 的顺序不能解决这个问题吧，这里不会影响 TabsNav 内部计算初始显示位置的时机

Comment (Saber-Kurama) [reply packages/web-vue/components/tabs/tabs.tsx]: 原本是打算用 flush: 'sync' 或者immediate:true来确保tabs的watch比activeIndex的watch先执行，在用vite写的demo的时候发现切换顺序也能让tabs的watch先执行。不过webpack中却不是这样的。保证watch的顺序还是得用flush: 'sync' 或者immediate:true

Comment (Saber-Kurama) [reply packages/web-vue/components/tabs/tabs.tsx]: 不过虽然可以让tabs的watch先执行，但是在activeIndex中的watch获取getsize中的值是wrapperLength.value 和  maxOffset.value不正确的

Comment (Saber-Kurama) [reply packages/web-vue/components/tabs/tabs.tsx]: activeIndex 中watch的`nextTick`我直接换成了`setTimeout`，也可以`watch([tabs,activeIndex]`判断`tabs`发生修改才用`setTimeout`。

Comment (Saber-Kurama) [reply packages/web-vue/components/tabs/tabs.tsx]: 另外就是直接设置`activeIndex ` 为最后一项的时候，`next`按钮并未置灰 。`offset.value`值比`maxOffset.value`小了几像素

**Pull request closed without merging**